src/core/server/saved_objects/migrations/types.ts

```diff
@@ -26,7 +26,7 @@ import { SavedObjectsMigrationLogger } from './core/migration_logger';
  *
  * @example
  * ```typescript
- * const migrateProperty: SavedObjectMigrationFn = (doc, { log }) => {
+ * const migrateProperty: SavedObjectMigrationFn<MyUnmigratedAttributes, MyMigratedAttributes> = (doc, { log }) => {
```

Comment:
we should update the example to show how to construct a copy so that the output is typed different to the input. Since this is only for type safety we don't need to make a deep copy, so it might be worth adding a comment to the example to highlight this.